File: goon_github/repositories.py
import logging

from github.Repository import Repository

from goon_github.github_client import github_client

log = logging.getLogger(__name__)


def get_all_repos() -> list[Repository]:
    try:
        repos = list(github_client.get_user("kjblanchard").get_repos())
        return repos
    except Exception as e:
        log.error(f"Error listing repositories: {e}")
        return []

File: goon_github/repositories.go
package goon_github

import (
	"log"

	"github.com/google/go-github/v60/github"
)

func GetAllRepos() []*github.Repository {
	if GithubClient == nil {
		log.Println("GitHub client not initialized")
		return nil
	}
	repos, _, err := GithubClient.Repositories.List(GetContext(), "kjblanchard", nil)
	if err != nil {
		log.Printf("Error listing repositories: %v", err)
		return nil
	}
	return repos
}
